viewing_party/party.py

```diff
@@ -1,23 +1,123 @@
 # ------------- WAVE 1 --------------------
 
+from hashlib import new
+from re import M
+from collections import Counter
+from viewing_party.helper_functions import *
+
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1, USER_DATA_3
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        new_movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data = {
+        "watched": [movie]
+    }             
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {
+        "watchlist": [movie]
+    }             
+    return user_data
+
+def watch_movie(user_data, title):    
+    new_watchlist = []
+    # iterates over watchlist. moves value in arg title from watchlist to 
+    # watched if found. generates a new watchlist comprised of unwatched movies 
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watched"].append(movie)
+        else:
+            new_watchlist.append(movie)
+    user_data["watchlist"] = new_watchlist
+    return user_data
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total_of_all_ratings = 0
+    movies_watched = len(user_data["watched"])
+    average = 0.0
+    if movies_watched > 0:
+        for movie in user_data["watched"]:
+            total_of_all_ratings += movie["rating"]
+        average = total_of_all_ratings / movies_watched
+    return average
+
+# imported Counter() from collections to use in this function
+def get_most_watched_genre(user_data):
+    dict_of_genres = Counter()
+    popular_genre = ""
+    if len(user_data["watched"]) > 0:

```

Comment:
If you invert the logic of this function with the guard clause, then this could be:
```python
if not user_data["watched"]: 
    return None. 
```

Then you can get rid of the if and else block and proceed with the logic to find the most watched genre.